### ISSUE TITLE
CASSCF gives wrong configurations but good energy

### ISSUE THREAD

**Issue by NastaMauger** [pyscf/pyscf#2406]
Hello,

I am trying to get familiar with PySCF and its different features. As a learning exercise, I am computing the energy of Be2 with different levels of theory. 

Using RHF and CAS(2,2), my PySCF output (energies and CI components) are identical with Orca software. However, when I move to CAS(4,4), I obtain different results with the two software packages. 

Here is my PySCF output:

```
CASSCF energy = -29.1572876405198
CASCI E = -29.1572876405198  E(CI) = -2.54343142415466  S^2 = 0.0000000
** Largest CI components **
  [alpha occ-orbitals] [beta occ-orbitals]            CI coefficient
  [0 1]                [0 1]                           0.983490117534
  [0 2]                [0 2]                          -0.119934378319
```

As you might know, only a CAS(4,4) can begin to correctly describe s-p hybridization. Therefore, these CI vectors do not make sense.

On the other hand, ORCA gives:

```
ROOT   0:  E=     -29.1572857274 Eh
      0.96807 [     5]: 0220 
      0.01440 [     1]: 2020 
      0.00657 [     0]: 2200 
      0.00542 [     7]: 0022 
      0.00506 [     6]: 0202 
```

I am a bit puzzled by PySCF's behavior, as it gives the correct energy (both codes give E = -29.157285 Ha), but PySCF's CI vectors and configurations are incorrect as I increased the CAS space. Since the energies are the same, I do not believe changing the convergence criterion will make a difference.

Here my input file:
[be_cas.txt](https://github.com/user-attachments/files/16924918/be_cas.txt)

Any help would be appreciated.
Best regards

**Comment by tberkel** [comment]
PySCF is giving the CI coefficients, Orca is giving the square of the
coefficients. And PySCF is only printing the largest coefficients.

**Comment by NastaMauger** [comment]
Hello,

I know that ORCA and PySCF print the coefficients differently; otherwise, I would not have succeeded for CAS(2,2). Moreover, the highest CI coefficients and configurations do not match. My input file prints all of them, but I only provided the largest in the discussion to make it easier to read. 

If you look closely at the highest coefficients given by both software, they do not match. PySCF gives 2200 with a value of 0.96725, while ORCA gives 0220 with a CI coefficient of 0.96807. The configurations (and to some extent the CI coefficients) do not match.
This is a weird behavior.
Best regards

**Comment by 1234zou** [comment]
Hi, @NastaMauger , you need to uncomment `#casscf.natorb = True` in your PySCF input file, and then you will find the square of the largest CI coefficient is 0.96808.

**Comment by NastaMauger** [comment]
Hello,

I was able to resolve my issue, which turned out to be twofold:

- First, the energies I obtained with ORCA using CASSCF were based on incorrect guess orbitals, leading to an overestimation of the 0220 configuration coefficient. Using the correct guess orbitals, or following advice from @1234zou, resolved this issue.

- The second issue was related to the basis set. Although I used the same basis set in both software, I noticed some discrepancies in the final MO energies. It turns out that directly setting up the basis set using `gto.parse` and the _aug-cc-pvdz_ basis from basis set exchange (NWChem format) resolved the issue. I'm not sure why this happened, but this approach fixed the problem.

Thank you so much for your time and replies—they were very helpful!
